src/api_encoder.rs

```diff
@@ -279,13 +280,14 @@ pub fn decode(data: &[u8]) -> Result<(KnownType, Bytes), DecodingErr> {
     Ok((tp, decoded))
 }
 
-fn split_prefix(data: &[u8]) -> Result<(String, Bytes), DecodingErr> {
-    if data.len() < 3 || data[2] != (b'_') {
+fn split_prefix(data: &str) -> Result<(String, Bytes), DecodingErr> {
+    let bytes = data.as_bytes();
+    if data.len() < 3 || bytes[2] != (b'_')  {
         Err(DecodingErr::MissingPrefix)?;
     }
 
-    let pfx = String::from_utf8(data[0..2].to_vec()).map_err(|_| DecodingErr::InvalidPrefix)?;
-    let payload = data[3..].to_vec();
+    let pfx = String::from_utf8(bytes[0..2].to_vec()).map_err(|_| DecodingErr::InvalidPrefix)?;
```

Comment:
Why not use the input string here? Does the borrow checker cry?

Reply:
Do you mean why convert it via `data.as_bytes`?
Because if we use things like `str.chars().nth()` it's O(n) complexity and there isn't a really good way to do substrings.
In our case we know this is an ASCII string and this code should be safe. 
I haven't done any benchmarking though